<launch>
  <arg name="robot_name_1" default="robot1" />
  <arg name="robot_name_2" default="robot2" /> 
  <arg name="robot_name_3" default="robot3" /> 

  <arg name="robot1_x" default="8.0"/>
  <arg name="robot1_y" default="-3.3"/>
  <arg name="robot1_a" default="0"/>

  <arg name="robot2_x" default="8.0"/>
  <arg name="robot2_y" default="5.2"/>
  <arg name="robot2_a" default="0"/>

  <arg name="robot3_x" default="-8.0"/>
  <arg name="robot3_y" default="-3.3"/>
  <arg name="robot3_a" default="0.0"/>


  <group ns="$(arg robot_name_1)/map_merge">
    <param name="init_pose_x"   value="$(arg robot1_x)"/>
    <param name="init_pose_y"   value="$(arg robot1_y)"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="$(arg robot1_a)"  />
  </group>

  <group ns="$(arg robot_name_2)/map_merge">
    <param name="init_pose_x"   value="$(arg robot2_x)"/>
    <param name="init_pose_y"   value="$(arg robot2_y)"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="$(arg robot2_a)"  />
  </group>

  <group ns="$(arg robot_name_3)/map_merge">
    <param name="init_pose_x"   value="$(arg robot3_x)"/>
    <param name="init_pose_y"   value="$(arg robot3_y)"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="$(arg robot3_a)"  />
  </group>

  <node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge" output="screen">
    <param name="robot_map_topic" value="map"/>
    <param name="robot_namespace" value="robot"/>
    <param name="merged_map_topic" value="map"/>
    <param name="world_frame" value="map"/>
    <param name="known_init_poses" value="true"/>
    <param name="merging_rate" value="30"/>
    <param name="discovery_rate" value="30"/>
    <param name="estimation_rate" value="30"/>
    <param name="estimation_confidence" value="1.0"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg robot_name_1)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg robot_name_1)/map 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg robot_name_2)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg robot_name_2)/map 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg robot_name_3)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg robot_name_3)/map 100"/>

  <node pkg="multi_bot_navigation" type="map_pub.py" name="map_pub" />
  <node pkg="multi_bot_navigation" type="multi_navi.py" name="multi_navi" />

  <node pkg="rviz" type="rviz" name="rviz" required="true"
        args="-d $(find multi_bot_navigation)/rviz/slam.rviz"/>

</launch>